### PR TITLE
Add HttpReqeuest GetEncodedUrl and GetDecodedUrl extensions.

### PR DESCRIPTION
**Pull request by Tratcher**
#341 
@davidfowl @Eilon

### CODE REVIEW

File: src/Microsoft.AspNet.Http.Extensions/UriHelper.cs [hunk 1]
```diff
@@ -71,5 +71,27 @@ public static string Encode(Uri uri)
...
+
+        /// <summary>
+        /// Returns the combine components of the request URL in a fully un-escaped form (except for the QueryString)
+        /// suitable only for display. This format should not be used in HTTP headers or other HTTP operations.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        public static string GetDecodedUrl(this HttpRequest request)
+        {
+            return request.Scheme + "://" + request.Host.Value + request.PathBase.Value + request.Path.Value + request.QueryString.Value;
```
Comment (Eilon) [inline src/Microsoft.AspNet.Http.Extensions/UriHelper.cs]: Does all this concatenation work when some sections are missing? E.g. no query string or empty path etc.?

**Comment by Eilon** [comment]
Also commented in the bug:

> We should re-discuss the names of these. Those the current names are technically correct, I don't believe they will lead to correct usage. The doc comments explain what's what, but I'm sure they'll be often ignored. I recommend perhaps changing  GetDecodedUrl()  to be something more like  GetDisplayUrl() , or perhaps even getting rid of it entirely.

**Comment by Tratcher** [comment]
Renamed GetDecodedUrl to GetDisplayUrl.

File: src/Microsoft.AspNet.Http.Extensions/UriHelper.cs [hunk 2]
```diff
@@ -71,5 +71,27 @@ public static string Encode(Uri uri)
                 return uri.GetComponents(UriComponents.SerializationInfoString, UriFormat.UriEscaped);
             }
         }
+
+        /// <summary>
+        /// Returns the combine components of the request URL in a fully escaped form suitable for use in HTTP headers
```
Comment (Eilon) [inline src/Microsoft.AspNet.Http.Extensions/UriHelper.cs]: `combined`

File: src/Microsoft.AspNet.Http.Extensions/UriHelper.cs [hunk 3]
```diff
@@ -71,5 +71,27 @@ public static string Encode(Uri uri)
...
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        public static string GetEncodedUrl(this HttpRequest request)
+        {
+            return Encode(request.Scheme, request.Host, request.PathBase, request.Path, request.QueryString);
+        }
+
+        /// <summary>
+        /// Returns the combine components of the request URL in a fully un-escaped form (except for the QueryString)
```
Comment (Eilon) [inline src/Microsoft.AspNet.Http.Extensions/UriHelper.cs]: `combined`

**Comment by Eilon** [comment]
:shipit: after the comment fix.

**Pull request merged**